server/use_cases/add_service_commitments.py

```diff
@@ -46,3 +75,99 @@ def check_time_overlap(shifts):
                 shifts[i].shift_end > shifts[j].shift_start):
                 return True
     return False
+
+# test cases
+if __name__ == "__main__":
+    from unittest.mock import Mock
+
+    # valid shift ID (successful case)
+    def test_valid_shift_commitment():
+        shifts_repo = Mock()
+        commitments_repo = Mock()
+
+        valid_shift = Mock()
+        valid_shift.get_id.return_value = (
+            "mocked-valid-shift-id"
+        )
+        shifts_repo.get_shifts.return_value = [valid_shift]
+
+        valid_commitment = Mock()
+        valid_commitment.service_shift_id = "mocked-valid-shift-id"
+
+        valid_commitment.to_dict.return_value = (
+            {"_id": "commitment-id"}
+        )
+
+        result = add_service_commitments(commitments_repo, shifts_repo, (
+            [valid_commitment])
+        )
+
+        assert result[0]["success"] is True
+        assert result[0]["service_commitment_id"] == "commitment-id"
+        print(f"Valid shift ID -> {result}\n")
+
+    # invalid shift ID (unsuccessful case)
+    def test_invalid_shift_commitment():
+        shifts_repo = Mock()
+        commitments_repo = Mock()
+
+        shifts_repo.get_shifts.return_value = []
+
+        invalid_commitment = Mock()
+        invalid_commitment.service_shift_id = "this-id-does-not-exist"
+
+        invalid_commitment.to_dict.return_value = (
+            {"_id": "commitment-id"}
```

Comment:
When we pass a service commitment to add_service_commitments use case, the commitment does not have an id.